packages/manager/src/features/CloudPulse/Alerts/CreateAlert/Criteria/DimensionFilterField.tsx

```diff
@@ -0,0 +1,186 @@
+        label: val,
+        value: val,
+      }));
+    }
+    return [];
+  };
+
+  return (
+    <Grid container data-testid={`${name}-id`} gap={2}>
+      <Grid item md={3} xs={12}>
+        <Controller
+          render={({ field, fieldState }) => (
+            <Autocomplete
+              onChange={(
+                _,
+                newValue: { label: string; value: string },
+                operation
+              ) => {
+                handleDataFieldChange(newValue, operation);
+              }}
+              value={
+                dataFieldOptions.find(
+                  (option) => option.value === field.value
+                ) ?? null
+              }
+              data-testid="Data-field"
+              disabled={dataFieldDisabled}
+              errorText={fieldState.error?.message}
+              label="Data Field"
+              onBlur={field.onBlur}
+              options={dataFieldOptions}
+              placeholder="Select a Data field"
+            />
+          )}
+          control={control}
+          name={`${name}.dimension_label`}
+        />
+      </Grid>
+      <Grid item md={2} xs={12}>
+        <Controller
+          render={({ field, fieldState }) => (
+            <Autocomplete
+              onChange={(
+                _,
+                newValue: { label: string; value: DimensionFilterOperatorType },
+                operation
+              ) => {
+                field.onChange(
+                  operation === 'selectOption' ? newValue.value : null
+                );
+              }}
+              value={
+                DimensionOperatorOptions.find(
+                  (option) => option.value === field.value
+                ) ?? null
+              }
+              data-testid="Operator"
+              errorText={fieldState.error?.message}
+              label="Operator"
+              onBlur={field.onBlur}
+              options={DimensionOperatorOptions}
+            />
+          )}
+          control={control}
+          name={`${name}.operator`}
+        />
+      </Grid>
+      <Grid item md={3} xs={12}>
+        <Box display={'flex'} gap={2}>
+          <Controller
+            render={({ field, fieldState }) => (
+              <Autocomplete
+                isOptionEqualToValue={(option, value) =>
+                  option.value === value.value
+                }
+                onChange={(
+                  _,
+                  selected: { label: string; value: string },
+                  operation
+                ) => {
+                  field.onChange(
+                    operation === 'selectOption' ? selected.value : null
+                  );
+                }}
+                value={
+                  valueOptions().find(
+                    (option) => option.value === field.value
+                  ) ?? null
+                }
+                data-testid="Value"
+                disabled={dimensionFieldWatcher === null}
```

Comment:
@santoshp210-akamai , is there any specific reason we check only for null,

it can be **disabled={!dimensionFieldWatcher}**